tests/test_redis_cache.py

```diff
@@ -238,3 +238,47 @@ def add_basic_get(arg1, arg2):
     r2_3_4, v2_3_4 = add_basic_get(3, 4)
 
     assert r_3_4 == r2_3_4 and v_3_4 == v2_3_4
+
+
+def test_same_name_method(cache):
+    class A:
+        @staticmethod
+        @cache.cache()
+        def static_method():
+            return 'A'
+    
+    class B:
+        @staticmethod
+        @cache.cache()
+        def static_method():
+            return 'B'
+    
+    A.static_method()
+    B.static_method()
+
+    # If there was a key conflict, the results should be the same
+    assert A.static_method() != B.static_method()
+
+
+def test_same_name_inner_function(cache):
+    def a():
+        @cache.cache()
+        def inner_function():
+            return 'A'
+        
+        inner_function() # Store the value in the cache
+        return inner_function
+    
+    def b():
+        @cache.cache()
+        def inner_function():
+            return 'B'
+
+        inner_function()
+        return inner_function
+    
+    first_func = a()
+    second_func = b()
+
+    # If there was a key conflict, the results should be the same
+    assert first_func() != second_func()
```

Comment:
Can you run these methods twice instead of just once? The first time won't cache. Let's also verify the keys.